proto/v1/compliance.proto

```diff
@@ -0,0 +1,80 @@
+// Copyright 2019 The Grafeas Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package grafeas.v1;
+
+import "google/protobuf/timestamp.proto";
+import "third_party/grafeas/proto/v1/common.proto";
+import "third_party/grafeas/proto/v1/cvss.proto";
+import "third_party/grafeas/proto/v1/package.proto";
+
+option go_package = "github.com/grafeas/grafeas/proto/v1/grafeas_go_proto";
+option java_multiple_files = true;
+option java_package = "io.grafeas.v1";
+option objc_class_prefix = "GRA";
+
+message ComplianceNote {
+  // The OS and config versions the benchmark applies to.
+  repeated grafeas.v1.Version version = 2;
+  // The title that identifies this compliance check.
+  string title = 3;
+  // A description about this compliance check. 
+  string description = 4;
+  // A rationale for the existence of this compliance check.
+  string rationale = 5;
+  // A description of remediation steps if the compliance check fails.
+  string remediation = 6;
+  // A compliance check that is a CIS benchmark.
+  message CisBenchmark {
+    int profile_level = 1;
+    grafeas.v1.Severity severity = 2;
+  }
+  oneof compliance_type {
+    CisBenchmark cis_benchmark = 7;
+  }
+  // Serialized scan instructions with a predefined format.
+  bytes scan_instructions = 8;
+}
+
+// Describes the CIS benchmark version that is applicable to a given OS and 
+// os version. 
+message Version {
+  // The CPE URI (https://cpe.mitre.org/specification/) this benchmark is
+  // applicable to.
+  string cpe_uri = 1;
+  // The version of the benchmark. This is set to the version of the OS-specific
+  // CIS document the benchmark is defined in.
+  string version = 2;
+}
+
+// An indication of how severe a violation of a compliance check is.
+enum Severity {
+  UNSPECIFIED = 0;
+  LOW = 1;
+  MEDIUM = 2;
+  HIGH = 3;
+  CRITICAL = 4;
+}
+
+message ComplianceOccurrence {
```

Comment:
There be a "string id = 1;" here too as per the design.

Reply:
I omitted it because the Occurrence proto has a way to set id: https://github.com/grafeas/grafeas/blob/master/proto/v1/grafeas.proto#L194

Reply:
Had a discussion offline, and we are agreeing to use a different struct / message to store the Id.